source/agora/common/crypto/ECC.d

```diff
@@ -103,6 +135,20 @@ public struct Scalar
         return this.data.toString();
     }
 
+    ///
+    unittest
+    {
+        Scalar s = Scalar.random();
+        char[] buff1;
+        char[] buff2;
+        scope dg1 = (const(char)[] data) { buff1 ~= data; };
+        scope dg2 = (const(char)[] data) { buff2 ~= data; };
+
+        s.toString(dg1, PrintMode.Obfuscated);
+        s.toString(dg2, PrintMode.Clear);
+        assert(buff1 != buff2);
```

Comment:
This is not really a good test, because it's not actually testing the output, only that it's different. It should test explicitly whether a private key is hidden. So what I suggest is:

- generate a random scalar
- print it out manually
- then hardcode it back into the first line in this test
- assert for both print modes exactly with the expected string output

Reply:
I updated `unittest`s. What do you think?